dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/builders/ProgramBuilder.java

```diff
@@ -84,6 +86,13 @@ public void setEntryPointId(String id) {
         entryPointId = id;
     }
 
+    public void setArch(Arch arch) {
+        if (this.arch != null) {
+            throw new ParsingException("Illegal attempt to override memory model");
+        }
+        this.arch = arch;
```

Comment:
`Preconditions.checkState(this.arch == null)`

Reply:
I think this is fine because we specifically throw a `ParsingException`. Your approach would give an `IllegalStateException`.